gobblin-service/src/main/java/org/apache/gobblin/service/monitoring/KafkaJobStatusMonitor.java

```diff
@@ -182,16 +182,14 @@ private static void modifyStateIfRetryRequired(org.apache.gobblin.configuration.
     }
   }
 
-  private static org.apache.gobblin.configuration.State mergedProperties(org.apache.gobblin.configuration.State jobStatus,
-      List<org.apache.gobblin.configuration.State> states) {
-    Properties mergedProperties = new Properties();
+  private static org.apache.gobblin.configuration.State mergeState(org.apache.gobblin.configuration.State state,
```

Comment:
Might be worth adding a javadoc: "Merge states based on precedence. The state instance in the 1st argument reflects the more recent state of a job (and is thus, given higher priority) compared to the 2nd argument."